src/azure-cli/azure/cli/command_modules/acr/check_health.py

```diff
@@ -72,11 +72,60 @@ def _subprocess_communicate(command_parts, shell=False):
     return output, warning, stderr, succeeded
 
 
+# Retrieve blob identified by digest - GET {url}/v2/{name}/blobs/{digest}
+def _get_blob(login_server, repository_name, digest):
+    return 'https://{}/v2/{}/blobs/{}'.format(login_server, repository_name, digest)
+
+# Checks inputted image or tag for possible matches
+def _check_image_match(cmd,
+                       registry_name,
+                       repository_name,
+                       login_server,
+                       username,
+                       password,
+                       image):
+    from difflib import get_close_matches
+    from .manifest import _get_manifest_path
+    from .repository import(
+        acr_repository_show_tags,
+        _obtain_data_from_registry)
+
+    if 'sha256' not in image:
+        tags = acr_repository_show_tags(
+            cmd,
+            registry_name=registry_name,
+            repository=repository_name,
+            orderby=None,
+            username=username,
+            password=password)
+        # Get a list of tags that have at least 80% similarity with the inputted image name
+        matches = get_close_matches(image, tags, n=1, cutoff=0.8)
+
+        if matches:
+            raise CLIError("Tag '{}' not found. Did you mean to input '{}'?".format(image, matches[0]))
+        return
+
+    # If digest is used, get all digests
+    image_digest_list = _obtain_data_from_registry(
```

Comment:
same here why not just pull the information from the service? I think this whole section is overcomplicating the issue. You can make that request regardless of Tag or Digest being the image format and get the image information

Reply:
@rosanch Yes I guess I don't need to check for tag or digest, it can just be one request. I think I was trying to customize the return message according to what was inputted but it's unnecessary

Reply:
that's all good. But that should be a step we do ONLY if we didn't find the image correct? If the input was a valid there's not need to add suggestions.

Reply:
Correct, the _check_image_match only gets called when the image data is not found.